lib/rex/exploitation/vbsobfuscate.rb

```diff
@@ -0,0 +1,141 @@
+# -*- coding: binary -*-
+
+module Rex
+  module Exploitation
+    #
+    # VBScript obfuscation library
+    #
+    class VBSObfuscate
+      # The VBScript code that this obfuscator will transform
+      attr_accessor :code
+
+      # Saves +code+ for later obfuscation with #obfuscate!
+      #
+      # @param code [#to_s] the code to obfuscate
+      # @param opts [Hash] an options hash
+      def initialize(code = nil, _opts = {})
+        self.code = code
+      end
+
+      # @return [String] the (possibly obfuscated) code
+      def to_s
+        @code
+      end
+
+      # Append +str+ to the (possibly obfuscated) code
+      def <<(str)
+        @code << str
+      end
+
+      # Obfuscate VBScript code.
+      #
+      # @option iterations        [Integer] number of times to run the obfuscator on this code (1)
+      # @option strip_whitespace  [Boolean] removes unnecessary whitespace from the output code (true)
+      # @option dynamic_execution [Boolean] dynamically execute obfuscated code with Execute (true)
+      #
+      # @return [self]
+      def obfuscate!(iterations: 1, strip_whitespace: true, dynamic_execution: true)
+        raise(ArgumentError, 'code must be present') if @code.nil?
+        raise(ArgumentError, 'iterations must be a positive integer') unless iterations.integer? && iterations.positive?
+
+        obfuscated = @code.dup
+
+        iterations.times do
+          # Normalize line endings and strip leading/trailing whitespace
+          if strip_whitespace
+            obfuscated.gsub!(/\r\n/, "\n")
+            obfuscated = obfuscated.lines.map(&:strip).reject(&:empty?).join("\n")
```

Comment:
Maybe a misunderstanding on my part - i assumed `strip_whitespace` would remove whitespace, but this seem to actually just be 'normalize'? 🤔 

If the intent is stripping, would this work (not tested)

```suggestion
            obfuscated = obfuscated.lines.map(&:strip).reject(&:empty?).join(';')
```

Or if the intent is normalization maybe renaming the option could be good for clarity

```diff
-          if strip_whitespace
+          if normalize_whitespace
```

Reply:
> Maybe a misunderstanding on my part - i assumed `strip_whitespace` would remove whitespace, but this seem to actually just be 'normalize'? 🤔

It removes lines which contain only whitespace and "strips" leading and trailing whitepace on all lines:

```ruby
# irb
irb(main):001> ' asdf asdf asdf '.strip
=> "asdf asdf asdf"
irb(main):002> 
```


> Or if the intent is normalization maybe renaming the option could be good for clarity

To me, "normalize" seems equally as vague as "strip" here, but I'll change it.


> If it's stripping, would this work (not tested)

No. VBS does not support command concatenation with `;` - only new lines.